### DISCUSSION TITLE
Seqeuncer上でSkeletalMeshにバインドされてるKawaiiPysicsに関して

### DISCUSSION THREAD

**Discussion by nihonjung** [pafuhana1213/KawaiiPhysics#98] [Q&A] [answered]
現在使ってるプロジェクトに参加してますが
①KawaiiPysicsはPreRollやPostRollでの抑え込みの影響は受けないように見えます。
その認識で正しいでしょうか？それとも使い方が間違ってるのでしょうか？
②KawaiiPysicsのAlphaの値をSeqeuncerでTrackで制御、Alphaの値もしくはOn/Off、したい場合はエンジンを改造して
　トラックを実装する方法しかないでしょうか？もしそうならどこのコードをどのようにいじればいいかのヒントをいただけますでしょうか？→多分プログラマーの仕事になると思うので方法を提示するときに交渉資料として出したいです。
よろしくお願いします。

**Comment by pafuhana1213** [comment, marked as answer]
> KawaiiPysicsはPreRollやPostRollでの抑え込みの影響は受けないように見えます。その認識で正しいでしょうか？それとも使い方が間違ってるのでしょうか？

手元で簡単なABP・Sequencerで試してみたのですが、PreRoll, PostRoll共に効いているように思います。KawaiiPhysicsがというよりもエンジン側の仕様ですが、キャラクタのABPやPhysicsの空回しに関しては様々な箇所にPreRoll・PostRoll 設定をしたり、Visibleの制御、非表示時の処理の有効化などを行う必要があります。過去に資料にまとめたことがあるので参考になると思います。テストの際はABPのTickにてPrintStringなどを仕込み、まずはPreRoll・PostRollでABPが動作しているのかをチェックするのがオススメです
https://www.docswell.com/s/EpicGamesJapan/5XP4YZ-UE4_GCC19_Sequencer_LatestInfoAndTips#p112
![image](https://github.com/pafuhana1213/KawaiiPhysics/assets/8957600/dcccfad5-58b1-4495-8159-2ba71f1131ef)

なお、いわゆる物理の空回し機能は以前より導入できればと思っていますが時間の関係で手を付けられていない状況です。初回時に FAnimNode_KawaiiPhysics::EvaluateSkeletalControl_AnyThread 内の処理を複数回走らせればそれっぽくなるとは考えています。


> ②KawaiiPysicsのAlphaの値をSeqeuncerでTrackで制御、Alphaの値もしくはOn/Off、したい場合はエンジンを改造してトラックを実装する方法しかないでしょうか？もしそうならどこのコードをどのようにいじればいいかのヒントをいただけますでしょうか？→多分プログラマーの仕事になると思うので方法を提示するときに交渉資料として出したいです。よろしくお願いします。

KawaiiPhysicsのAlphaに Expose to Cinematic を有効にしたプロパティを指定し、
<img width="562" alt="image" src="https://github.com/pafuhana1213/KawaiiPhysics/assets/8957600/29d7b7dd-a393-4f1a-85ca-2fd97f53283d">
そのAlphaをSequencer上で制御すれば実現可能です。
![image](https://github.com/pafuhana1213/KawaiiPhysics/assets/8957600/56a77d59-1722-4d87-8484-853fb5e7a89b)

**Comment by nihonjung** [comment]
返事が遅れて申し訳ありません。
有益な情報ありがとうございます。
プロジェクトのプログラマと情報共有して何が問題か検証してみます。
今後ともよろしくお願いします。

2023年8月12日(土) 22:02 pafuhana1213 ***@***.***>:
